mlir/include/mlir/Dialect/Bufferization/Transforms/Passes.h

```diff
@@ -167,6 +174,10 @@ struct BufferResultsToOutParamsOpts {
     return true;
   };
 
+  /// Allocation function; used to allocate a memref.
+  /// If this is empty, memref.alloc is used
+  std::optional<AllocationFn> allocationFn;
```

Comment:
Could this be avoided by adding a `buildAlloc` interface method to `AllocationOpInterface`? Maybe the transformation could also supported mixed alloc ops then>

Reply:
I'm not sure that would work actually. wouldn't you need an instance of an `AllocationOpInterface` to call the method?  In this context there is no such instance, since we are creating a new one. Maybe i'm missing something. Also I do think I prefer this since it mirrors alloc and copy customization in bufferization.

Reply:
Oh, you only have a memref SSA value, right?

Reply:
yah, this is used in `updateCalls` for allocating at the call site and uses the arg types to allocate